sls-versions/src/main/java/com/palantir/sls/versions/MatchResult.java

```diff
@@ -43,7 +45,14 @@ public int groupAsInt(int group) {
                 throw new NumberFormatException();
             }
 
-            return Integer.parseUnsignedInt(string, groupStart, groupEnd, RADIX);
+            int integer = Integer.parseUnsignedInt(string, groupStart, groupEnd, RADIX);
+            if (integer < 0) {
+                throw new SafeIllegalStateException(
+                        "Can't parse segment as integer as it overflowed",
+                        SafeArg.of("string", string),
+                        SafeArg.of("segment", string.substring(groupStart, groupEnd)));
```

Comment:
pedantic nit: We may want to extract this to a `integerOverflow(String,int,int);` function to cut negative test data out of hot paths for better inlining